javalin-plugins/javalin-swagger-plugin/src/main/kotlin/io/javalin/openapi/plugin/swagger/SwaggerPlugin.kt

```diff
@@ -35,14 +36,17 @@ open class SwaggerPlugin @JvmOverloads constructor(private val configuration: Sw
             routingPath = app.cfg.routing.contextPath,
             basePath = configuration.basePath
         )
+        /** Register handler for swagger ui */
+        app.get(configuration.uiPath, swaggerHandler, *configuration.roles)
 
-        val swaggerWebJarHandler = SwaggerWebJarHandler(
-            swaggerWebJarPath = configuration.webJarPath
-        )
-
-        app
-            .get(configuration.uiPath, swaggerHandler, *configuration.roles)
-            .get("${configuration.webJarPath}/*", swaggerWebJarHandler, *configuration.roles)
+        /** Register webjar handler if and only if there isn't already a [SwaggerWebJarHandler] at configured route */
+        val routes = app.javalinServlet().matcher.findEntries(HandlerType.GET, "${configuration.webJarPath}/*")
+        if(routes.isNotEmpty() && routes.first().handler !is SwaggerWebJarHandler){
```

Comment:
It looks like the handler is registered only if there's an entry that matches `${configuration.webJarPath}/*` and the first result is not `SwaggerWebJarHandler`. Am I missing sth? The test checks if response `isNotBlank`, but I think it's just 404 with some default message. It's late and I'm really tired after 🏂, so that's it from me today - I'll try to take a look at it tomorrow/in the weekend.

Reply:
Thanks for the comment. Indeed I missed a  `isEmpty` in that condition there.

I added an additional test case to check whether the (new) solution works with more than a shell `Javalin` and in my opinion, this works as intended and doesn't add any side-effects. Or do you see any issues with this approach?